code/cycler/src/wattrex_battery_cycler/mid/mid_dabs/mid_dabs.py

```diff
@@ -0,0 +1,393 @@
+            raise error
+
+    def __update_source_load_status(self, status: CyclerDataAllStatusC):
+        if status.source.value > 0 :
+            status.pwr_dev = status.source
+        elif status.load.value > 0:
+            status.pwr_dev = status.load
+        elif status.source.value < 0:
+            status.pwr_dev = status.source
+        elif status.load.value < 0:
+            status.pwr_dev = status.load
+        else:
+            status.pwr_dev = status.source
+
+    def update(self, gen_meas: CyclerDataGenMeasC, ext_meas: CyclerDataExtMeasC,#pylint: disable= too-many-branches
+               status: CyclerDataAllStatusC) -> None:
+        """Update the data from the hardware sendind the corresponding messages.
+        Update the variables of the class with the data received from the device.
+        Depending on the device type, the data will be updated in a way or another.
+        """
+        res = None
+        for dev_type, dev_id in zip(self.device_type, self.dev_id):
+            if dev_type is CyclerDataDeviceTypeE.BISOURCE:
+                res: DrvEaDataC = self.bisource.get_data()
+                status.pwr_dev = CyclerDataDeviceStatusC(error= res.status.error_code,
+                                                        dev_id= dev_id)
+            elif dev_type is CyclerDataDeviceTypeE.SOURCE:
+                res: DrvEaDataC = self.source.get_data()
+                status.source = CyclerDataDeviceStatusC(error= res.status.error_code,
+                                                dev_id= dev_id)
+            elif dev_type is CyclerDataDeviceTypeE.LOAD:
+                res: DrvRsDataC = self.load.get_data()
+                status.load = CyclerDataDeviceStatusC(error= res.status.error_code,
+                                                dev_id= dev_id)
+            elif dev_type is CyclerDataDeviceTypeE.METER:
+                res: DrvBkDataC = self.meter.get_data()
+                res = res.__dict__
+                for att_name in self.mapping_meter:
+                    setattr(ext_meas,att_name, getattr(res, att_name))
+            elif dev_type is CyclerDataDeviceTypeE.EPC:
+                msg_elect_meas = self.epc.get_elec_meas(periodic_flag= True)
+                msg_temp_meas = self.epc.get_temp_meas(periodic_flag= True)
+                msg_mode: DrvEpcDataC  = self.epc.get_mode()
+                epc_status = self.epc.get_status()
+                status.pwr_dev = CyclerDataDeviceStatusC(error= epc_status.error_code,
+                                                        dev_id= dev_id)
+                gen_meas.voltage = msg_elect_meas.ls_voltage
+                gen_meas.current = msg_elect_meas.ls_current
+                gen_meas.power   = msg_elect_meas.ls_power
+                status.pwr_mode = msg_mode.mode
+                if self.mapping_epc is not None:
+                    for key in self.mapping_epc.keys():
+                        if key == 'hs_voltage':
```

Comment:
Check is mandatory?

Reply:
Just check if temp is in key to get data from temp measure or from electrical